### PR TITLE
fix: Init release channels ui

### PR DESCRIPTION
**Pull request by adityachoudhari26**
## Release Notes

- **New Features**
  - Introduced components for managing release channels, including `ReleaseBadgeList`, `Overview`, `ReleaseChannelDrawer`, `ReleaseChannelDropdown`, `ReleaseFilter`, `CreateReleaseChannelDialog`, and `ReleaseChannelsTable`.
  - Added navigation for "Release channels" in the `DeploymentNavBar`.
  - Enhanced functionality for creating, updating, and deleting release channels through new API routes.
  - Implemented a custom hook for managing release channel state and navigation.

- **Bug Fixes**
  - Improved validation and error handling in the `ReleaseConditionDialog`.

- **Documentation**
  - Updated relevant sections to include new components and their functionalities.

### CODE REVIEW

File: apps/webservice/src/app/[workspaceSlug]/_components/release-channel-drawer/ReleaseFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import { ReleaseConditionBadge } from "../release-condition/ReleaseConditionBadge";
+import { ReleaseConditionDialog } from "../release-condition/ReleaseConditionDialog";
+import { ReleaseBadgeList } from "../ReleaseBadgeList";
+
+type ReleaseFilterProps = {
+  releaseChannel: SCHEMA.ReleaseChannel;
+};
+
+const getFinalFilter = (filter?: ReleaseCondition) =>
+  filter != null ? (isEmptyCondition(filter) ? undefined : filter) : undefined;
```
Comment (jsbroks) [inline apps/webservice/src/app/[workspaceSlug]/_components/release-channel-drawer/ReleaseFilter.tsx:26]:
```suggestion
const getFinalFilter = (filter?: ReleaseCondition) =>
  filter && !isEmptyCondition(filter) ? filter : undefined;
```

File: apps/webservice/src/app/[workspaceSlug]/_components/release-channel-drawer/Overview.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+              <FormLabel>Description</FormLabel>
+              <FormControl>
+                <Textarea {...field} />
+              </FormControl>
+            </FormItem>
+          )}
+        />
+
+        <Button type="submit">Save</Button>
+      </form>
```
[the remark this answers is not in the document]
Comment (jsbroks) [reply apps/webservice/src/app/[workspaceSlug]/_components/release-channel-drawer/Overview.tsx:80]: I like this

**Review by jsbroks — APPROVED** [review]

File: packages/auth/src/utils/rbac.ts [hunk 3]
```diff
@@ -98,6 +99,24 @@ const getReleaseScopes = async (id: string) => {
...
+    .where(eq(releaseChannel.id, id))
+    .then(takeFirst);
+
+  return [
+    { type: "releaseChannel" as const, id: result.release_channel.id },
+    { type: "deployment" as const, id: result.deployment.id },
+    { type: "system" as const, id: result.system.id },
+    { type: "workspace" as const, id: result.workspace.id },
+  ];
+};
```
[the remark this answers is not in the document]
Comment (adityachoudhari26) [reply packages/auth/src/utils/rbac.ts:118]: the scope is already added in this file, you hallucinating

File: packages/api/src/router/deployment.ts [hunk 4]
```diff
@@ -51,8 +56,103 @@ const latestActiveReleaseSubQuery = (db: Tx) =>
...
+              and(
+                eq(release.deploymentId, channel.deploymentId),
+                releaseMatchesCondition(ctx.db, filter),
+              ),
+            )
+            .then(takeFirst)
+            .then((r) => r.count);
+          return { ...channel, total };
+        });
+        return Promise.all(promises);
```
[the remark this answers is not in the document]
Comment (adityachoudhari26) [reply packages/api/src/router/deployment.ts:131]: required due to the limitation of the `releaseMatchesCondition`

**Pull request merged**